### PR TITLE
CD-167630 Add Admin API for topic management

### PR DESCRIPTION
**Pull request by johnny-lai**
- [x] @johnwu96822 
- [x] @abdollar

### CODE REVIEW

File: lib/rdkafka/admin_operations.rb [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+
+      # Read response
+      cnt_p = Rdkafka::Bindings::SizePtr.new
+      ptr = Rdkafka::Bindings.rd_kafka_CreateTopics_result_topics(event_p, cnt_p)
+      response = Rdkafka::Bindings::TopicResult.from_native_array(ptr, cnt_p[:value]).first
+      response.raise_if_error
+
+      return nil
+    ensure
+      Rdkafka::Bindings.rd_kafka_event_destroy(event_p) if event_p
```
Comment (abdollar) [inline lib/rdkafka/admin_operations.rb]: @johnny-lai maybe force event_p = nil so it doesn't get used by accident

Comment (johnny-lai) [reply lib/rdkafka/admin_operations.rb]: I am trying to freeing `event_p` here if it is not `nil`.

File: lib/rdkafka/rdlist.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+module Rdkafka
```
Comment (johnny-lai) [inline lib/rdkafka/rdlist.rb]: Unused file. Will remove

File: lib/rdkafka/admin_operations.rb (hunk 1 above)
Comment (abdollar) [reply lib/rdkafka/admin_operations.rb]: 👍

**Comment by abdollar** [comment]
👍

**Comment by johnny-lai** [comment]
The Ruby 2.5 is timing out. The other versions are green though.

File: lib/rdkafka/bindings.rb [hunk 3]
```diff
@@ -279,5 +280,177 @@ class TopicPartitionList < FFI::Struct
...
+    )
+
+    attach_function :rd_kafka_AdminOptions_new, [:pointer, AdminOptions], :pointer
+    attach_function :rd_kafka_AdminOptions_set_request_timeout, [:pointer, :int, :pointer, :size_t], :int
+    attach_function :rd_kafka_AdminOptions_set_operation_timeout, [:pointer, :int, :pointer, :size_t], :int
+    attach_function :rd_kafka_AdminOptions_destroy, [:pointer], :void
+
+    ResourceType = enum(
+      :RD_KAFKA_RESOURCE_UNKNOWN, 0,
+      :RD_KAFKA_RESOURCE_ANY, 1,
```
Comment (johnwu96822) [inline lib/rdkafka/bindings.rb]: How come the `AdminOptions` enum don't need the numbers??

**Comment by johnwu96822** [comment]
👍 One small question

File: lib/rdkafka/bindings.rb (hunk 3 above)
Comment (johnny-lai) [reply lib/rdkafka/bindings.rb]: So the numbers here are not needed.
But I decided to basically copy rdkafka.h: https://github.com/edenhill/librdkafka/blob/master/src/rdkafka.h#L5899

vs https://github.com/edenhill/librdkafka/blob/master/src/rdkafka.h#L5260

**Comment by johnny-lai** [comment]
Fixed issue where it crashes if it fails to find a response within the timeout.

**Pull request merged**